### PR TITLE
Update the minimum size of ViewportContainer

### LINKED ISSUE #19252: ViewportContainer can't change size smaller than initial Viewport size
**Godot version:**
3.1.dev.custom_build.46bab3a

**OS/device including version:**

**Issue description:**
I set Viewport and ViewportContainer size = [1024, 600]
I resize Viewport smaller than [1024, 600] and ViewportContainer still can't resize smaller than [1024, 600]
If over [1024, 600] that OK

**Steps to reproduce:**

**Minimal reproduction project:**
https://www.dropbox.com/s/tc9pjq78bf9pdkz/testViewport.zip?dl=0

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #19252

### CODE REVIEW

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -3685,6 +3689,14 @@ bool SubViewport::is_using_xr() {
...
 void SubViewport::set_size(const Size2i &p_size) {
 	_set_size(p_size, _get_size_2d_override(), Rect2i(), _stretch_transform(), true);
+#ifdef TOOLS_ENABLED
+	if (get_parent()) {
+		SubViewportContainer *c = Object::cast_to<SubViewportContainer>(get_parent());
+		if (c) {
+			c->minimum_size_changed();
+		}
+	}
+#endif
```
Comment (Xrayez) [inline scene/main/viewport.cpp:3699]: What is the reason for making this work in editor builds only? I already see bug reports mentioning that `SubViewportContainer` isn't resized/changed in `release` builds in the future. 😮

I think `TOOLS_ENABLED` should only be used if it's only relevant for the editor, like debug drawing.

Comment (KoBeWi) [reply scene/main/viewport.cpp:3699]: This only matters if you dynamically resize viewport during runtime, which sounds very rare (and we'd already have a report like that if anyone encountered this issue...). I just thought this is only relevant for the editor, although maybe there's no gain in disabling it 🤔

Comment (Xrayez) [reply scene/main/viewport.cpp]: > This only matters if you dynamically resize viewport during runtime

For example, game minimaps could be resized (like twice as big, or twice as small).

From what I've seen, people do manage to create non-game applications and editors, so things like that would be used relatively often there, but not necessarily for games I think. At most, think about built-in level editors.

File: scene/main/viewport.cpp [hunk 2]
```diff
@@ -3685,6 +3686,13 @@ bool SubViewport::is_using_xr() {
 
 void SubViewport::set_size(const Size2i &p_size) {
 	_set_size(p_size, _get_size_2d_override(), Rect2i(), _stretch_transform(), true);
+
+	if (get_parent()) {
+		SubViewportContainer *c = Object::cast_to<SubViewportContainer>(get_parent());
+		if (c) {
+			c->minimum_size_changed();
+		}
+	}
```
Comment (Xrayez) [inline scene/main/viewport.cpp:3695]:
```suggestion
	SubViewportContainer *c = Object::cast_to<SubViewportContainer>(get_parent());
	if (c) {
		c->minimum_size_changed();
	}
```
This should still work even if parent is null, AFAIK.

**Comment by akien-mga** [comment]
Needs a squash (and a review by @reduz).

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs another rebase I think, `minimum_size_changed` was removed IIRC.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
This probably warrants a dedicated backport PR for `3.x`.

**Pull request merged**